### PR TITLE
Feature/autoskip credits

### LINKED ISSUE #88: Missing "Automatically skip Credits"
I see "Automatically skip Intro" checkbox in plugin settings but don't see "Automatically skip Credits" or "Automatically press Next"
So I assumed that this checkbox would apply to both (Into and Credits), but it doesn't seems to be the case.

Intro is getting skipped automatically, however end credits still require you to press or hover your mouse on to "Next" button.

_Why have "Credit Detection" in scheduling tasks if there is no way to automatically skip it after it was detected?_

Can somebody please advise how to make it work? I just need button "Next" to be clicked automatically when it pops out during credits. 
 
I saw similar request in other places. But not for this plugin.  Is there another plugin that needs to be added to make this work, or could current [intro-skipper] include that?

### PR DESCRIPTION
**Pull request by phyzical**
Untested, very rough POC

the one thing i couldnt figure out is how to sepeate the dep on "AutoSkip" so that we can have two settings for credits and intros

as i cannot figure out how autoskip is initialized to make "type" part of the class as a whole.

if someone can point me in that direction ill change this pr to initlise two autoskip classes instead 

closes #88

### CODE REVIEW

**Comment by jumoog** [comment]
Thank you!

File: ConfusedPolarBear.Plugin.IntroSkipper/AutoSkip.cs [hunk 1]
```diff
@@ -115,90 +116,95 @@ private void UserDataManager_UserDataSaved(object? sender, UserDataSaveEventArgs
...
     }
 
     private void PlaybackTimer_Elapsed(object? sender, ElapsedEventArgs e)
     {
         foreach (var session in _sessionManager.Sessions)
         {
-            var deviceId = session.DeviceId;
-            var itemId = session.NowPlayingItem.Id;
-            var position = session.PlayState.PositionTicks / TimeSpan.TicksPerSecond;
+            Skip(sender, e,"intro", _sentIntroSeekCommandLock, !Plugin.Instance!.Intros, Plugin.Instance!.Configuration.AutoSkipIntroNotificationText)
```
Comment (jumoog) [inline ConfusedPolarBear.Plugin.IntroSkipper/AutoSkip.cs:132]: Please add missing ";"

File: ConfusedPolarBear.Plugin.IntroSkipper/AutoSkip.cs [hunk 2]
```diff
@@ -115,90 +116,95 @@ private void UserDataManager_UserDataSaved(object? sender, UserDataSaveEventArgs
...
+            Skip(sender, e,"credit", _sentCreditsSeekCommandLock, !Plugin.Instance!.Credits, Plugin.Instance!.Configuration.AutoSkipCreditsNotificationText)
```
Comment (jumoog) [inline ConfusedPolarBear.Plugin.IntroSkipper/AutoSkip.cs:133]: Please add missing ";"

**Review by jumoog — CHANGES_REQUESTED** [review]

File: ConfusedPolarBear.Plugin.IntroSkipper/AutoSkip.cs (hunk 1 above)
Comment (phyzical) [inline ConfusedPolarBear.Plugin.IntroSkipper/AutoSkip.cs:132]:
```suggestion
            Skip(sender, e,"intro", _sentIntroSeekCommandLock, !Plugin.Instance!.Intros, Plugin.Instance!.Configuration.AutoSkipIntroNotificationText);
```

File: ConfusedPolarBear.Plugin.IntroSkipper/AutoSkip.cs (hunk 2 above)
Comment (phyzical) [inline ConfusedPolarBear.Plugin.IntroSkipper/AutoSkip.cs:133]:
```suggestion
            Skip(sender, e,"credit", _sentCreditsSeekCommandLock, !Plugin.Instance!.Credits, Plugin.Instance!.Configuration.AutoSkipCreditsNotificationText);
```

**Comment by jumoog** [comment]
Did you try to compile the code before committing?

**Comment by phyzical** [comment]
no as i stated untested poc, was hoping for some guidance around refactoring it better to have two skippers inilisedisntead of merging them together inside, due to the setting for enabling autoskip being a core part of the class

**Comment by phyzical** [comment]
But if we dont care about that ill do a proper test i suppose this plugin will be redundant soon anwyay

**Comment by jumoog** [comment]
> But if we dont care about that ill do a proper test i suppose this plugin will be redundant soon anwyay

Unfortunately not! Media segements didn't make it into 10.9. and won't arrive until 10.10. At the current rate, that means in 2 years 😄

**Comment by AbandonedCart** [comment]
> Unfortunately not! Media segements didn't make it into 10.9.

The guy "writing it" is mostly stumbling through it and it seems like the team that needs to approve it is still bickering amongst themselves about how they want it to work. They had a whole mess of changes and were trying to rush him before they locked the code for 10.9, which allowed zero time for real testing. There was no way.

**Comment by phyzical** [comment]
guess what i mean is do we care if its kinda botched to pretend that credits are "intro" . looking at other parts that seems to be whats done so im leading toward the meh column

**Comment by jumoog** [comment]
Yes, that should refactor the code anyway

**Review by jumoog — DISMISSED** [review]

**Comment by phyzical** [comment]
once my credits processing is done, ill compile this and make sure it works as expected.

**Comment by jumoog** [comment]
The tricky part is always migrating the old data.

**Comment by phyzical** [comment]
we can only ditch he custom text for credits and reuse the intro text instead tbh

**Comment by AbandonedCart** [comment]
Somewhere in the original code, it was explained about why credits are actually based on intros.

**Comment by jumoog** [comment]
To be honest, I have not bothered with the code. I have seen that the old repo has been archived. Since I use the plugin anyway, I made a fork and made sure that the dependencies are updated.

**Comment by phyzical** [comment]
if i was to guess credits are intros... but in reverse 😆

**Comment by AbandonedCart** [comment]
No. It was something to do with how the credit placement is determined.

**Comment by jumoog** [comment]
only when the code no longer worked with 10.9 did i start looking at the code 😄

**Comment by AbandonedCart** [comment]
Still better than the project I am working on where the guy intentionally breaks code just to write something.

**Comment by jumoog** [comment]
> Still better than the project I am working on where the guy intentionally breaks code just to write something.

lol but Hacktoberfest is already over

**Comment by phyzical** [comment]
@jumoog 
image
any idea how to get around this build error?

**Comment by phyzical** [comment]
@jumoog 
image
any idea how to get around this build error?

**Comment by phyzical** [comment]
reopened fresh of master

**Pull request closed without merging**